### PR TITLE
Project maintainers represent Rails wherever they are known to discuss Rails

### PR DESCRIPTION
**Pull request by joeldrapper**
In this PR, I propose we update the “Scope” section of the Code of Conduct to make it clear that project maintainers are considered to represent Rails in any space where they are known to discuss Rails.

For example, it would be a Code of Conduct violation for a maintainer to express racist opinions on a Twitter account or blog where they also share Rails news. People follow maintainers in these spaces to learn about Rails. As such, maintainers should be conscious that they represent Rails in these spaces.

### CODE REVIEW

File: _pages/conduct.html [hunk 1]
```diff
@@ -47,7 +47,7 @@ <h3>Our Responsibilities</h3>
...
         <h3>Scope</h3>
-        <p>This Code of Conduct applies both within project spaces and in public spaces when an individual is representing the project or its community. Examples of representing a project or community include using an official project e-mail address, posting via an official social media account, or acting as an appointed representative at an online or offline event. Representation of a project may be further defined and clarified by project maintainers.</p>
+        <p>This Code of Conduct applies both within project spaces and in public spaces when an individual is representing the project or its community. Examples of representing a project or community include using an official project e-mail address, posting via an official social media account, or acting as an appointed representative at an online or offline event. For project maintainers, this applies to any space where they are known to discuss Rails. For example, it is not okay for a project maintainer to use the same Twitter account to share Rails news and express racist views. Representation of a project may be further defined and clarified by project maintainers.</p>
```
Comment (joeldrapper) [inline _pages/conduct.html:50]: Perhaps `are known` here is superfluous. Happy to remove it.

**Comment by gregmolnar** [comment]
The only problem is, who is going to decide what is racist? Because I assume your definition is different from mine. When I deal with someone, I don't care about that person's race because I believe, race doesn't make a difference, we are all the same human beings at the end of the day. 
So treating someone differently because they belong to a particular race is racist in my opinion, even when that person belongs to a minority and gets better treatment than someone who isn't from a group of minority just because of that.

**Comment by joeldrapper** [comment]
@gregmolnar Thanks for your feedback. I'm only proposing we extend the scope to include spaces where project maintainers talk about Rails. The Code of Conduct already covers racism and I'm not proposing any change to that.

**Comment by gdonald** [comment]
> who is going to decide what is racist?

Shouldn't the dictionary definition suffice?

**Comment by gregmolnar** [comment]
I knew you guys don't like to be blunt over the pond, but I am for Eastern Europe, we don't sugarcoat things over there. I know what this one is about. You read a blog post from DHH, you disagree with it, and now you want him to be removed from the Rails team.  
In my opinion, that blog post is not racist, and it follows the  Code of Conduct of Rails. You probably have a different opinion, which is fine(from the code of conduct: "Being respectful of differing viewpoints and experiences"), so I don't see why you think this change will make any difference.

**Comment by joeldrapper** [comment]
@gregmolnar I’m not here to argue about that. I leave it up to the Rails project maintainers to enforce the Code of Conduct as they see fit. My contribution here is about the scope because I hope we can agree that a Rails maintainer _does_ represent Rails when speaking in a space where they are known to discuss Rails.

**Comment by gregmolnar** [comment]
@joeldrapper if you ask me, that article is pretty racist. It only assumes "Black and Brown people" to experience racism. What about the rest? What about Asians?  Or gypsies in Europe? I could go on with the list, but I think that's not the real issue here. You guys are just doing the usual thing you do.

**Comment by gdonald** [comment]
> What about the rest? What about Asians? Or gypsies in Europe?

"Whataboutism" has entered the chat.

**Comment by joeldrapper** [comment]
@gregmolnar @gdonald Let’s agree to disagree about racism in this thread and instead focus on the specific scope contribution I’m proposing. I believe the Code of Conduct should cover spaces where project maintainers talk about Rails since, as Rails maintainers, they represent Rails in those spaces.

I haven’t changed any wording about what is and is not acceptable behaviour; I am only proposing to increase the scope to clarify that project maintainers represent Rails in spaces where they talk about Rails. You may disagree, but I think this is a reasonable point that I hope we can all agree on.

**Comment by gdonald** [comment]
> I don't see why you think this change will make any difference.

I'd be surprised if anything changes.  I fully expected this PR to be deleted long before now.

**Comment by joeldrapper** [comment]
> I don't see why you think this change will make any difference.

It makes a difference to how we talk about issues. "That’s out of scope" is a different conversation and a different position to "that doesn’t violate the Code of Conduct because..."

**Comment by gregmolnar** [comment]
@joeldrapper Why do you think that we need this change? Do you have any example of a maintainer breaking the Code of Conduct somewhere? Or would this change to just a preparation in case that ever happens? I would appreciate some honesty, I know that's a big ask.

**Comment by paracycle** [comment]
@joeldrapper While I understand why you want to make this change, I want to call out that the resulting implication is way wider than you make it out to be with your example in the change.

Let me present this unacceptable behaviour from the CoC:
> Trolling, insulting/derogatory comments, and personal or political attacks

If we widen the scope of conduct to all project maintainers' personal spaces where they also discuss Rails, then their personal Twitter accounts might be subject to CoC as well. If they are ever caught to be trolling another unrelated figure or making insulting/derogatory comments (let's say trolling or insulting Elon Musk on Twitter, for example), they also run the risk of violating the CoC.

I don't think that wide a scope is what anyone on the project wants, and is not doing anything for the inclusivity of the "project" itself.

**Comment by gdonald** [comment]
> I would appreciate some honesty, I know that's a big ask.

Your sugar coating is dripping all over the floor.

**Comment by karloscarweber** [comment]
Hi Friends

> If they are ever caught to be trolling another unrelated figure or making insulting/derogatory comments (let's say trolling or insulting Elon Musk on Twitter, for example), they also run the risk of violating the CoC.
	 
This is a great point by @paracycle, and why this should be adopted. What kind of community would accept trolling and insulting behaviour outside of their community but not inside? Be a jerk to, maybe PHP developers, but not Ruby developers?

I think that @gregmolnar is asking for some clarity:

> I would appreciate some honesty, I know that's a big ask.
	
@dhh writes racist, exclusionary, and incendiary things on his blog, thus creating a negative environment in the Rails community. This would make that not ok. Do YOU think it's racist or exclusionary? Do I? That doesn't matter in the context of this change.

**Comment by joeldrapper** [comment]
@paracycle thank you for your thoughtful comment and all you do for the Ruby community. 🙏 

I totally see where what you mean and appreciate the counter-example. I certainly don’t have all the answers, and this is a complex problem.

The way I would square it is insulting a public figure like Elon Musk would, in my opinion, not violate the spirit of the code of conduct in any way. But that’s not my decision to make, and I don’t know if it’s helpful for us to get drawn into specific examples.

Perhaps the wording in this pull request could be improved, but I think there's something here. When a Rails maintainer shares Rails information in a space, they do (at least to some extent) represent Rails in that space. That’s true whether the Code of Conduct recognises it or not.

For context, my concern is primarily about intolerant (e.g. sexist, racist, ableist) views being expressed alongside official Rails news to the extent that specific groups of people in the community who are there to follow the Rails news are made to feel unwelcome. I get that it's a thin line to walk, but I think we’d probably agree that, at some point, it’s not okay for someone to use their influence as a Rails maintainer to say harmful things and punch down.

**Comment by fxn** [comment]
My personal spaces are my personal spaces.

You know when companies force you to say "opinions my own"? Well, obviously opinions are my own, and my personal blog, my personal social accounts, etc. are personal and no CoC can tell me what to say there.

The fact that I blog about mechanical watches does no imply the core team has an interest on mechanical watches, or that they share my preference of mechanical vs quartz. Of course, the Rails project has no opinion on watches that you can transfer from my personal spaces.

In the Rails project we do not talk about watches, we talk about whether we like this Active Record API to be this or that way.

Definitely, personal spaces are out of the Rails CoC.

**Comment by matthewd** [comment]
Sorry, I really just don't think this is a viable standard to apply.

The example is specific, but it's clearly not up to the Code of Conduct to arbitrarily regulate everything that someone might say -- either the full Code of Conduct applies, or it does not. (i.e., this cannot be a "don't be racist" rule in isolation.)

The Project, through the Code of Conduct, regulates its own speech, and thus that of anyone speaking in an official capacity.

I don't want The Project to be entitled to an opinion on everything I say, everywhere. Whether I've been incidentally identified as a maintainer of Rails, or have incidentally mentioned a contextually relevant breaking change I'm aware of / a feature I've been working on and am excited about, does not change that.

I spend a large amount of my free time working on Rails; it comes up in conversation. And I live on the main branch; if we're drawing a broad circle, everything there is potentially "news".

Like every other member of Core, David's blog and twitter are not sources of official project news. You don't need to follow us for that purpose. On the rare occasion that we might post there about something news-worthy that we're personally excited about, it will be reflected through the official project news channels (twitter, newsletter, blog, release notes) as appropriate.

Moreover, the effect of such a policy would be felt most strongly by those maintainers who are not already comfortable having all their speech filtered through the lens of broad public discourse. To oblige someone to live their entire life in a professional setting, as penance for volunteering some of their free time to work on an open source project, is unreasonable.

Sometimes people say things that I disagree with. Sometimes people express opinions I'm not inclined to be respectful of. And sometimes people say things that just seem incredibly, or deliberately, naive. I don't think it is fair to prevent me from expressing those thoughts, in my personal capacity, in project-unaffiliated places -- and as tactlessly and/or unprofessionally as fits my mood.

I assume this limitation of reasonableness is why the Contributor Covenant explicitly constrains its scope here.

**Comment by matthewd** [comment]
I'm pretty sure the UI doesn't show specific attribution, so for the record: I've just marked a bunch of previous comments as off topic.

**Comment by joeldrapper** [comment]
Thanks for your input @matthewd. I understand your perspective on this.

Perhaps it is too thin of a line to walk and erring on the less restrictive side is for the best.

I do wish the maintainers would extend the spirit of the Code of Conduct into their own spheres, treating people with respect and making everyone feel welcome in the Rails community as a matter of personal choice. This has overwhelmingly been the case with only a few sad exceptions.

I don't have anything further to add to the conversation here.

Thanks again for your time @matthewd, @paracycle and @gregmolnar. 🙏

**Pull request closed without merging**